### PR TITLE
Remove SocketStreamHandleImpl.h of curl and soup ports

### PR DESCRIPTION
**Pull request by fujii**
#### 097ca320af8eecc7d67965238fd9342ab13fe946
<pre>
Remove SocketStreamHandleImpl.h of curl and soup ports
<a href="https://bugs.webkit.org/show_bug.cgi?id=254201">https://bugs.webkit.org/show_bug.cgi?id=254201</a>

Reviewed by NOBODY (OOPS!).

SocketStreamHandleImpl.h is included by SocketStreamHandleImpl.cpp and
WebSocketChannel.cpp. SocketStreamHandleImpl.cpp is used only by Cocoa
port. Moved SocketStreamHandleImpl.cpp into cf directory.
WebSocketChannel.cpp is used only by Cocoa port. Guarded the entire
file with PLATFORM(COCOA). Removed SocketStreamHandleImpl.h of curl
and soup ports.

* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/Modules/websockets/WebSocketChannel.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/SocketProvider.cpp:
(WebCore::SocketProvider::createWebSocketChannel):
* Source/WebCore/platform/Curl.cmake:
* Source/WebCore/platform/Soup.cmake:
* Source/WebCore/platform/network/cf/SocketStreamHandleImpl.cpp: Renamed from Source/WebCore/platform/network/SocketStreamHandleImpl.cpp.
* Source/WebCore/platform/network/curl/SocketStreamHandleImpl.h: Removed.
* Source/WebCore/platform/network/soup/SocketStreamHandleImpl.h: Removed.
</pre>
https://github.com/WebKit/WebKit/commit/097ca320af8eecc7d67965238fd9342ab13fe946

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/94 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/95 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/75 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/88 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/87 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/73 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/99 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/80 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/72 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/87 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/81 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/71 "3 failures") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/67 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/47 "Failed to checkout and rebase branch from PR 11748") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/85 "Built successfully") | | | 
| | | | |

### CODE REVIEW

File: Source/WebCore/SourcesCocoa.txt [hunk 1]
```diff
@@ -635,6 +635,7 @@ platform/mediastream/mac/WebAudioSourceProviderCocoa.mm
 platform/mock/MediaPlaybackTargetMock.cpp
 platform/mock/MediaPlaybackTargetPickerMock.cpp
 platform/mock/MockAudioDestinationCocoa.cpp
+platform/network/SocketStreamHandleImpl.cpp
```
Comment (mcatanzaro) [inline Source/WebCore/SourcesCocoa.txt]: platform/network/cf/

**Review by mcatanzaro — APPROVED** [review]
I would move the file as well to minimize potential for confusion. Anyway, looks fine.

**Comment by fujii** [comment]
Thank you for the review. But, https://bugs.webkit.org/show_bug.cgi?id=254179 landed most part of this change. I will rebase this patch and rename the title.

**Review by mcatanzaro — APPROVED** [review]
Still looks good.

Optional: I would try moving WebSocketChannel.cpp to a cocoa/ directory if you can.

**Comment by fujii** [comment]
> Optional: I would try moving WebSocketChannel.cpp to a cocoa/ directory if you can.

Yeah, it should be move. But, WebSocketChannel.h is still used somewhere. It needs some refactoring. I keep WebSocketChannel.cpp stayed next to WebSocketChannel.h at the moment.

**Comment by fujii** [comment]
Fixed by https://bugs.webkit.org/show_bug.cgi?id=254221

**Pull request closed without merging**